app/src/app/[lng]/cities/[cityId]/GHGI/[inventory]/page.tsx

```diff
@@ -1,12 +1,63 @@
 "use client";
-import { use } from "react";
+import { useEffect } from "react";
+import { useParams, useRouter } from "next/navigation";
+import { api } from "@/services/api";
+import InventoryPage from "@/components/shared/InventoryPage";
 
-import HomePage from "@/components/GHGIHomePage/HomePage";
+export default function CityInventoryPage() {
+  const router = useRouter();
+  const { cityId, inventory, lng } = useParams();
 
-export default function PrivateHome(props: {
-  params: Promise<{ lng: string }>;
-}) {
-  const { lng } = use(props.params);
+  const cityIdValue = Array.isArray(cityId) ? cityId[0] : cityId;
+  const inventoryValue = Array.isArray(inventory) ? inventory[0] : inventory;
+  const lngValue = Array.isArray(lng) ? lng[0] : lng;
 
-  return <HomePage lng={lng} isPublic={false} />;
+  // Get user info to check for default inventory
+  const { data: userInfo, isLoading: userInfoLoading } =
+    api.useGetUserInfoQuery();
+
+  // Get inventory data to validate if the inventory exists and user has access
+  const {
+    data: inventoryData,
+    error: inventoryError,
+    isLoading: inventoryLoading,
+  } = api.useGetInventoryQuery(inventoryValue!, { skip: !inventoryValue });
+
+  useEffect(() => {
+    if (userInfoLoading || inventoryLoading) return;
+
+    // If inventory doesn't exist or user doesn't have access, redirect appropriately
+    if (inventoryError || !inventoryData) {
+      if (userInfo?.defaultInventoryId) {
+        // Redirect to default inventory
+        router.replace(`/${lngValue}/${userInfo.defaultInventoryId}`);
+      } else if (userInfo?.defaultCityId) {
+        // Redirect to default city
+        router.replace(`/${lngValue}/cities/${userInfo.defaultCityId}`);
+      } else {
+        // Redirect to onboarding
+        router.replace(`/${lngValue}/onboarding`);
+      }
```

Comment:
I've seen this logic before in a different file on this PR. Might be good to extract to a reusable function/ hook so we can do this consistently across our application.